From e771154c37bc6fe9f87d08fde802125c02a241e3 Mon Sep 17 00:00:00 2001
From: Jed Brown <jed@jedbrown.org>
Date: Wed, 25 Apr 2018 14:16:55 -0600
Subject: [PATCH] Include <petsc/mpiuni/mpi.h> instead of requiring extra -I on
 command-line

This makes a PETSc --with-mpi=0 behave more predictably for people who
do not want to know about MPI.
---
 include/petscsys.h                                 | 6 +++++-
 src/benchmarks/streams/MPIVersion.c                | 2 --
 src/dm/examples/tests/ex42.c                       | 3 +--
 src/tao/leastsquares/examples/tutorials/chwirut2.c | 1 -
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/include/petscsys.h b/include/petscsys.h
index 6d893c0a8e..43e07d644a 100644
--- a/include/petscsys.h
+++ b/include/petscsys.h
@@ -127,7 +127,11 @@ void assert_never_put_petsc_headers_inside_an_extern_c(int); void assert_never_p
 #if !defined(OMPI_WANT_MPI_INTERFACE_WARNING)
 #  define OMPI_WANT_MPI_INTERFACE_WARNING 0
 #endif
-#include <mpi.h>
+#if defined(PETSC_HAVE_MPIUNI)
+#  include <petsc/mpiuni/mpi.h>
+#else
+#  include <mpi.h>
+#endif
 
 /*
    Perform various sanity checks that the correct mpi.h is being included at compile time.
diff --git a/src/benchmarks/streams/MPIVersion.c b/src/benchmarks/streams/MPIVersion.c
index 569e107b21..6fe8a7aeec 100644
--- a/src/benchmarks/streams/MPIVersion.c
+++ b/src/benchmarks/streams/MPIVersion.c
@@ -73,8 +73,6 @@ static double bytes[4] = {
   3 * sizeof(double) * N
 };
 
-#include <mpi.h>
-
 int main(int argc,char **args)
 {
   int            quantum, checktick(void);
diff --git a/src/dm/examples/tests/ex42.c b/src/dm/examples/tests/ex42.c
index 5e9d7a0200..216980c5b9 100644
--- a/src/dm/examples/tests/ex42.c
+++ b/src/dm/examples/tests/ex42.c
@@ -2,9 +2,8 @@
 
 static char help[] = "Test VTK Rectilinear grid (.vtr) viewer support\n\n";
 
-#include <mpi.h>
 #include <petscdm.h>
-#include "petscdmda.h"
+#include <petscdmda.h>
 
 /*
   Write 3D DMDA vector with coordinates in VTK VTR format
diff --git a/src/tao/leastsquares/examples/tutorials/chwirut2.c b/src/tao/leastsquares/examples/tutorials/chwirut2.c
index 2868aee2cf..7c94e4f1d0 100644
--- a/src/tao/leastsquares/examples/tutorials/chwirut2.c
+++ b/src/tao/leastsquares/examples/tutorials/chwirut2.c
@@ -9,7 +9,6 @@
 */
 
 #include <petsctao.h>
-#include <mpi.h>
 
 
 /*
